core/src/main/scala/swave/core/util/RingBuffer.scala

```diff
@@ -118,5 +118,17 @@ private[swave] final class RingBuffer[T](cap: Int) {
     java.util.Arrays.fill(array, null)
   }
 
+  /**
+    * Iterates the underlying elements of the buffer
+    *
+    */
+  def foreach[U](f: T => U): Unit = {
```

Comment:
This implementation is completely fine, readable and fast.
However, it contains a `var` that we could get rid off by moving to a `@tailrec def`.
If you want you could try rewriting the method in this way (just for the hack of it).

I've been trying to not use `while` and always use `@tailrec def`s in all my code for some time.
The benefit is that, once you get used to it, it becomes really natural and offers a number of benefits in more complex cases.

Admittedly, in simple cases like this a `while` is just as good a solution and a `@tairec def` doesn't really add any benefit (but has no drawbacks either!) apart from maybe keeping the overall code style a bit more consistent.